general_ledger/general_ledger.py

```diff
@@ -202,15 +241,24 @@ def _get_key_id(currentKey):
             else:
                 parties_subtitle = ''
 
-        company = fiscalyear.company
+        if data['company']:
+            company = Company(data['company'])
+        elif fiscalyear:
+            company = fiscalyear.company
+        else:
+            company = Company(Transaction().context.get('company', -1))
 
         parameters = {}
         parameters['company'] = company.rec_name
         parameters['company_vat'] = (company.party.tax_identifier
             and company.party.tax_identifier.code) or ''
         parameters['start_period'] = start_period and start_period or ''
         parameters['end_period'] = end_period and end_period or ''
-        parameters['fiscal_year'] = fiscalyear.rec_name
+        parameters['start_date'] = (start_date.strftime('%d/%m/%Y')
+            if start_date else '')
+        parameters['end_date'] = (end_date.strftime('%d/%m/%Y')
+            if end_date else '')
+        parameters['fiscal_year'] = fiscalyear.rec_name if fiscalyear else None
```

Comment:
For "parameters" could be better use an empty string '', not a None.

Reply:
Is jinja, not jasper

```
{% if fiscal_year %} .... {% endif %}
```

not equal to:

```
{% if fiscal_year == '' %} .... {% endif %}
```

Reply:
I know it's jinja. But I think for strings is better use:

parameters['fiscal_year'] = fiscalyear.rec_name if fiscalyear else ''

If it's necessary to check "fiscal_year" has information, you could do in jinja in various ways too. For example:

{% if fiscal_year|length %}

or

{% if fiscal_year != '' %}

Reply:
472821e327fd1fdf067d3be78c3fbcdf7448bed2